### PR TITLE
fix(auth): Delete user invalid state fixes

### PR DESCRIPTION
**Pull request by tylerjroach**
- [ ] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:* #2288

*Description of changes:*

*How did you test these changes?*
(Please add a line here how the changes were tested)

*Documentation update required?*
- [ ] No
- [ ] Yes (Please include a PR link for the documentation update)

*General Checklist*
- [ ] Added Unit Tests
- [ ] Added Integration Tests
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by gpanshu — DISMISSED** [review]

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/DeleteUserCognitoActions.kt [hunk 1]
```diff
@@ -22,31 +22,44 @@ import com.amplifyframework.statemachine.Action
...
-                            SignOutData(globalSignOut = false)
-                        )
-                    )
+                    val evt = DeleteUserEvent(DeleteUserEvent.EventType.ThrowError(e, true))
+                    dispatcher.send(evt)
                 } else {
-                    DeleteUserEvent(DeleteUserEvent.EventType.ThrowError(e))
+                    val evt = DeleteUserEvent(DeleteUserEvent.EventType.ThrowError(e, false))
+                    dispatcher.send(evt)
+                    val evt2 = AuthorizationEvent(AuthorizationEvent.EventType.ThrowError(e))
```
Comment (div5yesh) [inline aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/DeleteUserCognitoActions.kt:47]: Please add log events sent.

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/DeleteUserState.kt [hunk 2]
```diff
@@ -54,14 +53,21 @@ internal sealed class DeleteUserState : State {
...
                 is DeletingUser -> {
                     when (deleteUserEvent) {
-                        is DeleteUserEvent.EventType.SignOutDeletedUser -> StateResolution(UserDeleted())
+                        is DeleteUserEvent.EventType.UserDeleted -> {
+                            val action = deleteUserActions.initiateSignOut()
+                            StateResolution(UserDeleted(), listOf(action))
+                        }
                         is DeleteUserEvent.EventType.ThrowError -> {
-                            StateResolution(Error(deleteUserEvent.exception))
+                            if (deleteUserEvent.signOutUser) {
```
Comment (div5yesh) [inline aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/DeleteUserState.kt:61]: I don't see signOutUser property in the deleteUserEvent. Did you add in this PR?

Comment (tylerjroach) [reply aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/DeleteUserState.kt:61]: There is no sign out process in Delete User anymore. It proceeds from DeletingUser state to SigningOut state.

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt [hunk 3]
```diff
@@ -1734,40 +1735,35 @@ internal class RealAWSCognitoAuthPlugin(
...
+                        authZState is AuthorizationState.DeletingUser &&
+                                authZState.deleteUserState is DeleteUserState.Error -> {
+                            deleteUserException = authZState.deleteUserState.exception
+                        }
+                        authNState is AuthenticationState.SignedOut && authZState is AuthorizationState.Configured -> {
+                            sendHubEvent(AuthChannelEventName.USER_DELETED.toString())
+                            authStateMachine.cancel(listenerToken)
+                            onSuccess.call()
+                        }
+                        authZState is AuthorizationState.SessionEstablished && exception != null -> {
```
[the remark this answers is not in the document]
Comment (tylerjroach) [reply aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt:1755]: Its not set in exception, its set on every authState update if in AuthState.Configured.

**Review by gpanshu — APPROVED** [review]

**Review by div5yesh — APPROVED** [review]

**Pull request merged**